monitor/script.js

```diff
@@ -0,0 +1,220 @@
+            cumulativeLiquidate: [],
+          },
+          poolErrors: {
+            totalValueLockedUSD: [],
+            cumulativeSupplySideRevenueUSD: [],
+            cumulativeProtocolSideRevenueUSD: [],
+            cumulativeTotalRevenueUSD: [],
+            cumulativeDepositUSD: [],
+            cumulativeBorrowUSD: [],
+            cumulativeLiquidateUSD: [],
+            totalBorrowBalanceUSD: [],
+            totalDepositBalanceUSD: [],
+            outputTokenSupply: [],
+            outputTokenPriceUSD: [],
+          },
+          url: deploymentString,
+          protocolType: protocolType,
+        };
+      });
+    });
+  });
+
+  const fullCurrentQueryArray = ["query Status {"];
+  const fullPendingQueryArray = ["query Status {"];
+
+  const queryContents = `
+  subgraph
+  node
+  synced
+  health
+  fatalError {
+    message
+    handler
+  }
+  nonFatalErrors {
+    message
+    handler
+  }
+  chains {
+    network
+    chainHeadBlock {
+      number
+    }
+    earliestBlock {
+      number
+    }
+    latestBlock {
+      number
+    }
+    lastHealthyBlock {
+      number
+    }
+  }
+  entityCount`;
+
+  Object.keys(deployments).forEach((name) => {
+    fullCurrentQueryArray[fullCurrentQueryArray.length - 1] += `        
+          ${name
+        .split("-")
+        .join(
+          "_"
+        )}: indexingStatusForCurrentVersion(subgraphName: "messari/${name}") {
+            ${queryContents}
+          }
+      `;
+    fullPendingQueryArray[fullPendingQueryArray.length - 1] += `        
+        ${name
+        .split("-")
+        .join(
+          "_"
+        )}_pending: indexingStatusForPendingVersion(subgraphName: "messari/${name}") {
+          ${queryContents}
+        }
+    `;
+    if (fullCurrentQueryArray[fullCurrentQueryArray.length - 1].length > 80000) {
+      fullCurrentQueryArray[fullCurrentQueryArray.length - 1] += "}";
+      fullCurrentQueryArray.push(" query Status {");
+    }
+    if (fullPendingQueryArray[fullPendingQueryArray.length - 1].length > 80000) {
+      fullPendingQueryArray[fullPendingQueryArray.length - 1] += "}";
+      fullPendingQueryArray.push(" query Status {");
+    }
+  });
+  fullCurrentQueryArray[fullCurrentQueryArray.length - 1] += "}";
+  fullPendingQueryArray[fullPendingQueryArray.length - 1] += "}";
+
+  const currentStateDepos = deployments;
+  Object.keys(currentStateDepos).forEach((name) => {
+    deployments[name + "-pending"] = { ...deployments[name] };
+  });
+
+  const indexingStatusQueries = [
+    ...fullCurrentQueryArray,
+    ...fullPendingQueryArray,
+  ].map((query) =>
+    axios.post("https://api.thegraph.com/index-node/graphql", { query: query })
+  );
+  let indexData = [];
+  await Promise.all(indexingStatusQueries)
+    .then(
+      (response) =>
+      (indexData = response.map(
+        (resultData) => (resultData.data.data)
+      ))
+    )
+    .catch((err) => console.log(err));
+
+  indexData = { ...indexData[0], ...indexData[1] };
```

Comment:
@jihok 
https://github.com/messari/subgraphs/pull/826#discussion_r952162468
Before this logic, indexData is an array with two elements; an object of results from the current deployments query and an object of results from the pending deployments query. Before there was a loop here that would settle this into one object saved to indexData. This logic now condenses this into one line with a spread operator.